source/code/source/playfab/PlayFabCurlHttpPlugin.cpp

```diff
@@ -110,6 +112,27 @@ namespace PlayFab
         } // UNLOCK httpRequestMutex
     }
 
+    constexpr char requestIdHeaderKey[] = "X-RequestId:";
+    constexpr char whitespace[] = "\t\n\v\f\r ";
+    size_t HeaderCallback(char* buffer, size_t size, size_t nitems, void* userdata)
+    {
```

Comment:
I think you should add a check at the beginning of this callback that if the requestId is non-null then return early, to skip processing any other headers.